src/storage/src/hummock/iceberg_compactor_runner.rs

```diff
@@ -211,185 +403,74 @@ impl IcebergCompactorRunner {
-                            .load_manifest_list(
-                                committed_table.file_io(),
-                                committed_table.metadata(),
-                            )
-                            .await
-                            .map_err(|e| HummockError::compaction_executor(e.as_report()))?;
-
-                        for manifest_file in manifest_list
-                            .entries()
-                            .iter()
-                            .filter(|entry| entry.has_added_files() || entry.has_existing_files())
-                        {
-                            let manifest = manifest_file
-                                .load_manifest(committed_table.file_io())
-                                .await
-                                .map_err(|e| HummockError::compaction_executor(e.as_report()))?;
-                            let (entry, _) = manifest.into_parts();
-                            for i in entry {
-                                match i.content_type() {
-                                    iceberg::spec::DataContentType::Data => {
-                                        input_files.push(i.data_file().clone());
-                                    }
-                                    iceberg::spec::DataContentType::EqualityDeletes => {
-                                        unreachable!(
-                                            "Equality deletes are not supported in main branch"
-                                        );
-                                    }
-                                    iceberg::spec::DataContentType::PositionDeletes => {
-                                        unreachable!(
-                                            "Position deletes are not supported in main branch"
-                                        );
-                                    }
-                                }
-                            }
-                        }
+                let plan_stats = self.compact_with_plan(plan, &context).await?;
 
-                        input_files
-                    } else {
-                        vec![]
-                    }
-                };
+                tracing::info!(
+                    task_id = task_id,
+                    plan_index = plan_index,
+                    stat = ?plan_stats,
+                    "Completed compaction plan {}/{}",
+                    plan_index + 1,
+                    total_plan_count,
+                );
 
-                let _new_table = commit_manager
-                    .overwrite_files(data_files, input_files, MAIN_BRANCH)
-                    .await
-                    .map_err(|e| HummockError::compaction_executor(e.as_report()))?;
+                all_stats.push(plan_stats);
             }
 
-            Ok::<RewriteFilesStat, HummockError>(stats)
+            // Merge all statistics
+            let merged_stats = Self::merge_rewrite_stats(all_stats);
```

Comment:
It looks like we only merge the stats, but would generate multi snapshots during partial compaction.